project/frontend/src/containers/CreateCompetition/CreateCompetition.js

```diff
@@ -24,7 +24,28 @@ const CreateCompetition = (props) => {
   const onParticipantDelete = (participant) =>
     setParticipantList(participantList.filter((v) => v !== participant?.email));
 
+  function checkValidDate() {
+    const start = new Date(startDate);
+    const end = new Date(endDate);
+    const now = new Date();
+    const today = new Date(now.getFullYear(), now.getMonth(), now.getDate());
+    if (end < start) {
+      setDateErrorMessage(
+        "Invalid date selection! start date cannot be after end date."
+      );
+    } else if (start < today) {
+      setDateErrorMessage(
+        "Invalid date selection! start date cannot be before today."
+      );
+    } else {
+      setDateErrorMessage("");
+    }
+    return end > start && start >= today;
+  }
+
   const sendInfo = async () => {
+    if (!checkValidDate()) return;
```

Comment:
I think we should also check that `start` and `end` are not null. This would happen if they were constructed with an empty string, ie `new Date("") == null`

Reply:
I just realised that when the date is not selected it's not null but it is 'Invalid Date' and to check for an invalid date will takes a couple lines of code. What do you think about check for an empty string for the startDate and endDate state. Which is what I did here ea98b68d5d7ce4f3603cebf26cf01aa0fffcb413
![image](https://user-images.githubusercontent.com/75593542/107608225-36822c80-6ca0-11eb-9fc7-bd7b0fb20cce.png)